### DISCUSSION TITLE
Add dynamic token_audience

### DISCUSSION THREAD

**Discussion by yeyeric** [fastapi-users/fastapi-users#789] [Q&A]
Hello,

I was wondering if there was a "native" way in the package to include dynamic scope based on each user ?

Indeed, in the code in /authentication/jwt.py it's managed like this to generate the token:
`data = {"user_id": str(user.id), "aud": user.token_audience}`

I'd like to understand how I could send dynamic token_audience like user.token_audience ?

I've managed to do it by adding another class like this but was wondering if there was another way as it would feel logic to me that every scopes is bound to a user and evaluated at run time, not when we instantiate the JWTAuthentication class

```
class JWTAuthenticationCustom(JWTAuthentication):
    async def _generate_token(self, user: UserDB) -> str:
        data = {"user_id": str(user.id), "aud": user.token_audience}
        return generate_jwt(data, self.secret, self.lifetime_seconds)

```

Thanks

**Comment by frankie567** [comment]
In most cases, a JWT audience is used to identify "recipients that the JWT is intended for". So having it statically defined makes totally sense in the general case. 

If you need to have it dynamically defined following user properties, what you did looks good to me. You can even add your own properties in the payload, that's totally allowed.

NB: For questions like this, please favor [Discussions](https://github.com/fastapi-users/fastapi-users/discussions) over issues in the future 🙏
